### PR TITLE
feat: Implemented the filter logic on the nav links

### LINKED ISSUE #41: Feature: Set up filter logic
### Type of feature

Feature

### Current behavior

The filters on the page exist as empty links. These are meant filters based on their relevantly named label. 

Screen Shot 2021-12-26 at 8 48 25 AM

### Desired solution
Some sort of Filter State logic should get added, similar to the [LayoutToggle](https://github.com/open-sauced/hot/blob/226768ffbb8220e6888b8ed9f0a7044bba1e55cd/src/components/LayoutToggle.jsx#L3) (state). 

Label | How it should work 
-|-
Popular | Default API filter (by [most starred by open sauced](https://github.com/open-sauced/hot/blob/226768ffbb8220e6888b8ed9f0a7044bba1e55cd/src/data/hot.json#L7) users)
Upvoted | Not yet implemented #12 
Discussed | Filter by [issue count](https://github.com/open-sauced/hot/blob/226768ffbb8220e6888b8ed9f0a7044bba1e55cd/src/data/hot.json#L6)
Recent | Filter by lowest [avg_recency_score](https://github.com/open-sauced/hot/blob/226768ffbb8220e6888b8ed9f0a7044bba1e55cd/src/data/hot.json#L8)

### Additional context

This feature should apply a filter and re-order how the repos are listed on the page. This is a common feature and there are a number of examples out there. 

https://medium.com/suyeonme/react-how-to-implement-an-infinite-scroll-749003e9896a

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by hokagedemehin**
## What type of PR is this? (check all applicable)

- [x] ♻️ Refactor
- [x] ✨ Feature

## Description

This PR adds the filter feature on the secondary nav link so as to change the data displayed based on the link that is clicked. I will explain my thought process on how I went about completing this task

- I removed `SecondaryNav.astro` and created `SecondaryNav.jsx`. I did this because I wanted to use some react hooks to perform state management in the component
- Each nav list had a `data-name` tag with unique names of the link. I did this so as to implement conditional styling on the active link. I wanted to dynamically change the style of the active link so that when any of the links are clicked, it sets the given style that I needed on it and adjusts the rest
- In the `PostWrap.jsx` file , I added the `SecondaryNav.jsx` there instead of the `index.astro` file and also created two new components `GridDisplay.jsx` and `ListDisplay.jsx` to replace `PostGrid.jsx` and `PostList.jsx` respectively
-  Inside of the two new files is where the sorting was done based on the nav link that was clicked and also the `PostGrid.jsx` and `PostList.jsx` file was used here to render out the final data

## Related Tickets & Documents

Fixes #41 

#### Popular nav link screenshot (Default API filter)
Screenshot 2022-01-02 215511

#### Discussed nav link screenshot (Filter by issue count)
Screenshot 2022-01-02 215526

#### Recent nav link screenshot (Filter by lowest avg_recency_score)
Screenshot 2022-01-02 222929
## Added tests?

- [x] 🙅 no, because they aren't needed

## Added to documentation?

- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by hokagedemehin** [comment]
Hello @0-vortex 

Could you please review this PR and tell me what you think, if it meets the requirements of the issue?

🤞🏾🤞🏾🤞🏾

**Comment by 0-vortex** [comment]
> Hello @0-vortex
> 
> Could you please review this PR and tell me what you think, if it meets the requirements of the issue?
> 
> 🤞🏾🤞🏾🤞🏾

Checked the deploy preview and the functionality matches the issue description, will give the code a second look in the morning and get back to you 🍕 

Thank you for contributing ❤️

**Comment by hokagedemehin** [comment]
👀

**Comment by 0-vortex** [comment]
Follow up to the initial review: other than HTML comments I am not sure how the backend/pagination would look like. @bdougie is this ok to pass with local filtering?

**Comment by bdougie** [comment]
> Follow up to the initial review: other than HTML comments I am not sure how the backend/pagination would look like. @bdougie is this ok to pass with local filtering?

For now, this works since the API doesn't have pagination or anything. Quite honestly I problem I am going to punt on until we get live data on the page. 

My thought is to do sort and cache it from the API in local storage, but I have not done this personally. My experience is limited to paginated APIs like GitHub. I will open an issue on this once this is merged.

**Review by bdougie — CHANGES_REQUESTED** [review]
Remove the comment component and open the issue for testing filtering. Once that is done, this is good to merge.

Thanks for taking this on, just be sure to check if issues are assigned before starting the work in the future.

File: src/pages/index.astro [hunk 1]
```diff
@@ -49,7 +49,7 @@ let social_card = 'https://repository-images.githubusercontent.com/71359796/179e
 <!-- page -->
   <main>
     <PrimaryNav />
-    <SecondaryNav />
+    <!-- <SecondaryNav /> -->
```
[the remark this answers is not in the document]
Comment (hokagedemehin) [reply src/pages/index.astro]: I have removed the commented component and unused imports

File: src/components/ListDisplay.jsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      <div className=" container space-y-3 ">
+        {activeLink == "popular"
+          ? fetchedData.map((item, i) => <PostList data={item} key={i} />)
+          : activeLink == "upvoted"
+          ? fetchedData.map((item, i) => <PostList data={item} key={i} />)
+          : activeLink == "discussed"
+          ? fetchedData.sort((a, b) => b.issues - a.issues).map((item, i) => <PostList data={item} key={i} />)
+          : fetchedData
+              .sort((a, b) => a.avg_recency_score - b.avg_recency_score)
+              .map((item, i) => <PostList data={item} key={i} />)}
```
[the remark this answers is not in the document]
Comment (hokagedemehin) [reply src/components/ListDisplay.jsx:16]: I have opened an issue for this PR to be tested. it is here #64 

Please could you check if it is fine by you

File: src/components/SecondaryNav.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import React, { useState } from "react";
+
+const SecondaryNav = ({ activeLink, setActiveLink }) => {
+  // console.log(activeLink);
```
Comment (bdougie) [inline src/components/SecondaryNav.jsx:4]: Dead code should be deleted
```suggestion
```

File: src/components/SecondaryNav.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+              className={`${
+                activeLink === "popular" ? "bg-cheesyYellow rounded-xl text-grey " : " "
+              } hover:text-saucyRed transition-all duration-300 mr-3 p-2 sm:mr-11`}
+            >
+              Popular
+            </li>
+            <li
+              data-name="upvoted"
+              onClick={(e) => handleChange(e)}
+              // className="hover:text-saucyRed transition-all duration-300 mr-3 sm:mr-11"
```
Comment (bdougie) [inline src/components/SecondaryNav.jsx:27]: Dead code again
```suggestion
```

File: src/components/SecondaryNav.jsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+              className={`${
+                activeLink === "upvoted" ? "bg-cheesyYellow rounded-xl text-grey " : " "
+              } hover:text-saucyRed transition-all duration-300 mr-3 p-2 sm:mr-11`}
+            >
+              Upvoted
+            </li>
+            <li
+              data-name="discussed"
+              onClick={(e) => handleChange(e)}
+              // className="hover:text-saucyRed transition-all duration-300 mr-3 sm:mr-11"
```
Comment (bdougie) [inline src/components/SecondaryNav.jsx:37]:
```suggestion
```

File: src/components/SecondaryNav.jsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+              className={`${
+                activeLink === "discussed" ? "bg-cheesyYellow rounded-xl text-grey " : " "
+              } hover:text-saucyRed transition-all duration-300 mr-3 p-2 sm:mr-11`}
+            >
+              Discussed
+            </li>
+            <li
+              data-name="recent"
+              onClick={(e) => handleChange(e)}
+              // className="hover:text-saucyRed  transition-all duration-300 mr-3 sm:mr-11"
```
Comment (bdougie) [inline src/components/SecondaryNav.jsx:47]:
```suggestion
```

**Review by bdougie — COMMENTED** [review]
Best practice is to remove dead code instead of commenting it out. The ternary operators seem to be replacing these, and this comment is not providing any context that is needed.

**Pull request merged**